app.js

```diff
@@ -1,8 +1,123 @@
+const cupThreeEl = document.getElementById('cup-three');
+
+const ballEl = document.getElementById('ball');
+
+const guessButtonOneEl = document.getElementById('guess-button-one');
+const guessButtonTwoEl = document.getElementById('guess-button-two');
+const guessButtonThreeEl = document.getElementById('guess-button-three');
+
+const playAgainButtonEl = document.getElementById('play-again-button');
+const correctGuessesEl = document.getElementById('correct-guesses');
+const incorrectGuessesEl = document.getElementById('incorrect-guesses');
+const totalGuessesEl = document.getElementById('total-guesses');
+const resetButtonEl = document.getElementById('reset-button');
+
 
 // let state
+let correctGuesses = 0;
+let incorrectGuesses = 0;
+let totalGuesses = 0;
+
+playAgainButtonEl.disabled = true;
+
+const hidingPlaces = ['one', 'two', 'three'];
 
 // set event listeners 
+guessButtonOneEl.addEventListener('click', () => {
+    const hidingSpot = Math.floor(Math.random() * hidingPlaces.length);
+    const answer = hidingPlaces[hidingSpot];
+    handleGuess('one', answer);
+    toggleButtons();
+});
+
+guessButtonTwoEl.addEventListener('click', () => {
+    const hidingSpot = Math.floor(Math.random() * hidingPlaces.length);
+    const answer = hidingPlaces[hidingSpot];
+    handleGuess('two', answer);
+    toggleButtons();
+});
+
+guessButtonThreeEl.addEventListener('click', () => {
+    const hidingSpot = Math.floor(Math.random() * hidingPlaces.length);
+    const answer = hidingPlaces[hidingSpot];
+    handleGuess('three', answer);
+    toggleButtons();
+});
+
+playAgainButtonEl.addEventListener('click', () => {
+    playAgainButtonEl.disabled = true;
+    guessButtonOneEl.disabled = false;
+    guessButtonTwoEl.disabled = false;
+    guessButtonThreeEl.disabled = false;
+    resetCups();
+});
+
+resetButtonEl.addEventListener('click', () => {
+    resetScoreboard();
+    resetCups();
+
+    playAgainButtonEl.disabled = true;
+    guessButtonOneEl.disabled = false;
+    guessButtonTwoEl.disabled = false;
+    guessButtonThreeEl.disabled = false;
+});
+
   // get user input
-  // use user input to update state 
-  // update DOM to reflect the new state
+function handleGuess(userGuess, correctSpot) {
+    totalGuesses++;
+
+    if (correctSpot === 'one'){
+        ballEl.style.gridColumn = '1';
+        cupOneEl.style.marginTop = '-120px';
+    } else if (correctSpot === 'two'){
+        ballEl.style.gridColumn = '2';
+        cupTwoEl.style.marginTop = '-120px';
+    } else if (correctSpot === 'three'){
+        ballEl.style.gridColumn = '3';
+        cupThreeEl.style.marginTop = '-120px';
+    }
+
+    if (userGuess === correctSpot) {
+        correctGuesses++;
+    } else {
+        incorrectGuesses++;
+    }
+
+    // update DOM to reflect the new state
+    correctGuessesEl.textContent = correctGuesses;
+    incorrectGuessesEl.textContent = incorrectGuesses;
+    totalGuessesEl.textContent = totalGuesses;
+}
```

Comment:
Since you have state for correctGuesses and totalGuesses you can the value of the "incorrect" guesses can be inferred. This means you might be able to care about one less piece of state (and save an infinitesimal amount of memory). 
```js
incorrectGuessesEl.textContent = totalGuesses - correctGuesses;
```
If you infer the incorrect guesses this way elsewhere in the code you could also utilize a neat trick called short circuiting. The if/else statement on line 83 could be re-written to something like:

```js
(userGuess === correctSpot) && correctGuesses++;
```

Read about short circuiting the && operator here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND#short-circuit_evaluation

Reply:
Counterargument against using short circuiting as an `if` replacement.

https://www.codereadability.com/dont-use-short-circuiting-when-you-want-an-if-statement